pkg/storage/metastore/badger.go

```diff
@@ -0,0 +1,687 @@
+// Copyright 2021 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metastore
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	badger "github.com/dgraph-io/badger/v3"
+	"github.com/google/uuid"
+	"google.golang.org/protobuf/proto"
+
+	pb "github.com/parca-dev/parca/gen/proto/go/parca/metastore/v1alpha1"
+)
+
+// BadgerMetastore is an implementation of the metastore using the badger KV
+// store.
+type BadgerMetastore struct {
+	db *badger.DB
+}
+
+var _ ProfileMetaStore = &BadgerMetastore{}
```

Comment:
The compiler should generally be able to catch this when assigning the badger store to the metastore interface in the main.go within that switch case, shouldn't it? Generally not a fan of this pattern in Go.

Reply:
Ah, this is left from the start of the implementation. I agree once it's integrated it's pointless.